src/app/file-manager/services/login.service.ts

```diff
@@ -16,34 +18,52 @@ const googleConfig = {
 
 const googleOptions = {
   scope: ['email',
-          'profile',
-          'https://www.googleapis.com/auth/compute',
-          'https://www.googleapis.com/auth/devstorage.full_control'].join(' '),
+    'profile',
+    'https://www.googleapis.com/auth/compute',
+    'https://www.googleapis.com/auth/devstorage.full_control'].join(' '),
   accessType: 'offline'
 };
 
 @Injectable()
 export class LoginService {
   constructor(private electronService: ElectronService,
-              private store: Store<any>) {}
+    private store: Store<any>,
+    private firecloudService: FirecloudService) { }
 
-  public googleLogin(): Promise<void> {
-     if (this.electronService.isElectronApp) {
-      const promise = new Promise<void>((resolve, reject) => {
+  public googleLogin(): Promise<any> {
+    if (this.electronService.isElectronApp) {
+
+      return new Promise<any>((resolve, reject) => {
         this.electronService.ipcRenderer.send('google-oauth', googleConfig, googleOptions);
         this.electronService.ipcRenderer.on('sendRendererMessage', (event, props) => {
           SecurityService.setAccessToken(props.result.access_token);
-          this.store.dispatch(login());
-          resolve();
+          // Verifies if usr is registerd to FireCloud
+          this.firecloudService.getUserRegistrationStatus().subscribe(
+            registered => {
+              this.store.dispatch(login());
+              resolve();
+            },
+            notRegistered => {
+              SecurityService.removeAccessToken();
+              reject(notRegistered);
+            });
         });
-      });
-      return promise;
+      }
+      );
     } else {
       alert('Implement web login');
       return null;
     }
   }
 
+  setGoogleOptions(): Observable<any> {
+    return new Observable<any>(
```

Comment:
Do you need to make a new observable each time this method is called? Or could we make this object once and return the same one each time?

Reply:
This observable has been removed and replaced by a constructor call to configure google options.